web/docs/formats.md

```diff
@@ -15,18 +17,139 @@ Parsers and printers interact with their corresponding dual from a
+to huge schemas and imprecise results. Use with caution.
+
+This option must not be combined with `--selector` or `--raw --schema`.
+
+### `--schema <schema>` (Parsers)
+
+Explicitly set the output schema.
+
+If a schema with a matching name is installed, the result will have exactly
+all fields of the specified schema.
+Fields that are in the schema, but did not appear in the input will be null.
+Fields that appear in the input, but not in the schema will be discarded.
+
+If the given schema does not exist, this option only assigns the output schema name only.
+
+This option must not be combined with `--selector` or `--raw --merge`.
+
+### `--selector <field>[:<prefix>]` (Parsers)
+
+Similar to `--schema`, but use the value of the field specified in `<field>`
+as the schema name.
+
+If the optional `<prefix>` is specified, then the schema is prepended with a
+prefix. For example, the selector `event_type:suricata` with an event that has
+the field `event_type` set to the value `flow` looks for a schema named
+`suricata.flow`.
+
+This option must not be combined with `--merge` or `--schema`.
+
+### `--schema-only` (Parsers)
+
+If the an existing schema was given/selected, enabling this option will ensure that the output
+schema has *only* the fields from that schema. If the schema name is obtained via a `selector`
+and it does not exist, this has no effect.
+
+This option requires either `--schema` or `--selector` to be set.
+
+### `--unnest-separator <separator>` (Parsers)
+
+A delimiter that, if present in keys, causes values to be treated as values of
+nested records.
+
+A popular example of this is the [Zeek JSON](formats/zeek-json.md) format. It includes
+the fields `id.orig_h`, `id.orig_p`, `id.resp_h`, and `id.resp_p` at the
+top-level. The data is best modeled as an `id` record with four nested fields
+`orig_h`, `orig_p`, `resp_h`, and `resp_p`.
+
+Without an unnest separator, the data looks like this:
+
+```json
+{
+  "id.orig_h" : "1.1.1.1",
+  "id.orig_p" : 10,
+  "id.resp_h" : "1.1.1.2",
+  "id.resp_p" : 5
+}
+```
+
+With the unnest separator set to `.`, Tenzir reads the events like this:
+
+```json
+{
+  "id" : {
+    "orig_h" : "1.1.1.1",
+    "orig_p" : 10,
+    "resp_h" : "1.1.1.2",
+    "resp_p" : 5
+  }
+}
 ```
 
-When a printer contructs raw bytes, it sets a
+### `--raw` (Parsers)
+
+Use only the raw types that are native to the parsed format. Fields that have a type
+specified in the chosen schema will still be parsed according to the schema.
+
+For example, the JSON format has no notion of an IP address, so this will cause all IP addresses
+to be parsed as strings, unless the fields is specified to be an IP address by the schema.
+JSON however has numeric types, so those would be parsed.
+
+Use with caution.
+
+This option must not be combined with `--merge --schema`.
+
+### `--unique-selector` (Parsers)
+
+Only consider the selector fields value for batching.
+
+This option requires `--selector`.
```

Comment:
That is indeed the effect. I think we briefly talked about this.

This came out of the investigation into the suricata "slowdown" compared to the old implementation. Using just the selector field value obviously speeds up the signature computation immensely (de-facto removes it), so I considered this a worthwhile option.

I agree that re-using the `--merge` flag for this would be preferable, but because we originally decided to simplify the configuration space to a single "linear" choice (merge, precise, selector), that would have been quite a bit more work to implement.

In the end it turns out that we cant even enable it by default for suricata, because events may have nested subevents and other fields even if they have the same selector value.

Reply:
I see. What do we do with it now? It can't stay like this because of the bad option name.

Looks like it should be very easy to remove `--unique-selector`, remove the check for `--merge --selector`, rename `policy_selector::unique_selector` to `policy_selector::merge`, and set that depending on `--merge` in `multi_series_builder_argument_parser::get_policy`.

Or do you see any problem with that?